### ISSUE TITLE
nn from scratch

### ISSUE THREAD

**Issue by vvksh** [vvksh/learning_stuff#8]
Implement a NN from scratch for reference

**Comment by vvksh** [comment]
check out vvksh/nnfromscratch_mnist

**Comment by vvksh** [comment]
Training details:

for theory look at neuralnetworksanddeeplearning.com.

Main code calls `Network.SGD()`
     SGD: 
        ---> creates mini batches, calls `update_mini_batch` on each batches
      update_mini_batch:
        ---> loops through each tuple in the minibatch and calls backprop on each tuple
        ---> backprop returns weight and bias deltas, this method then updates the weights and biases
      backprop:
        ---> calls forward code and calculates gradients.
